cqf-fhir-utility/src/main/java/org/opencds/cqf/fhir/utility/r4/PackageHelper.java

```diff
@@ -159,11 +159,17 @@ public static BundleEntryComponent createEntry(Resource resource, boolean isPut)
             request.setMethod(HTTPVerb.PUT).setUrl(resourceType + "/" + resource.getIdPart());
         } else {
             request.setMethod(HTTPVerb.POST).setUrl(resourceType);
-            if (hasUrl(resource)) {
-                var url = getUrl(resource);
-                if (hasVersion(resource)) {
+        }
+        if (hasUrl(resource)) {
+            var url = getUrl(resource);
+            if (hasVersion(resource)) {
+                entry.setFullUrl(url + "|" + getVersion(resource));
+                if (!isPut) {
                     request.setIfNoneExist(String.format("url=%s&version=%s", url, getVersion(resource)));
-                } else {
+                }
+            } else {
+                entry.setFullUrl(url);
+                if (!isPut) {
```

Comment:
Please make sure these changes are also made in the Dstu3 and R5 versions of this class.